src/rdiff_backup/selection.py

```diff
@@ -630,7 +630,7 @@ def glob_get_filename_sf(self, filename, include):
 		globbing characters are used.
 
 		"""
-        if not filename.startswith(self.prefix):
+        if not filename.startswith(self.prefix + b"/"):
```

Comment:
if you do something like the following, it should work:

```python
self.prefixindex = self.rpath.index  # inserted after line 88 to avoid splitting ourselves

[...]

fileindex = tuple([x for x in filename.split(b"/") if x])

# are the first elements of the path the same?
if fileindex[:len(prefixindex)] != self.prefixindex:
            raise FilePrefixError(filename)
index = fileindex[len(self.prefixindex):]
```

Actually, you should pack these lines in a method that returns the index based on filename, because you can reuse it at lines 438 to 441 of the same file.